### PR TITLE
Delta progress

### PR DESCRIPTION
**Pull request by cgwalters**
One bugfix, one enhancement, and work on fixing delta progress.

### CODE REVIEW

**Comment by cgwalters** [comment]
Ok, pushed some more work here.  This clearly wants some tests, which is going to require a bit of (long overdue) work.  So I'll keep WIP until then.

File: src/libostree/ostree-repo-static-delta-compilation.c [hunk 1]
```diff
@@ -1003,11 +1003,7 @@ generate_delta_lowlatency (OstreeRepo                       *repo,
       if (!check_object_world_readable (repo, from_checksum, &from_world_readable, cancellable, error))
         goto out;
       if (!from_world_readable)
-        {
-          g_hash_table_iter_steal (&hashiter);
-          g_hash_table_add (new_reachable_regfile_content, (char*)from_checksum);
-          continue;
-        }
+        continue;
```
Comment (dbnicholson) [inline src/libostree/ostree-repo-static-delta-compilation.c]: Took me a couple reads through this code again, but I agree this is right. There's no reason you'd want to include the old object in the delta. The rest of the code will DTRT as long as this unreadable object is kept out of the bsdiffs and rollsums to that the old object is not required on the pull side when applying the delta.

**Comment by cgwalters** [comment]
Removing WIP.  It's clear to me now that we really need a "realistic repo" test which is what I was going for with `test-pull-many.sh`.  But running it currently is expensive; we'd need to refactor basically all of the tests to somehow derive from that - basically we only want to build the repo once.  Or maybe we move to having it be static, and stash it in S3, and simply require fetching it from the network?

**Comment by giuseppe** [comment]
that sounds like a good idea, should be possible though to build it locally as well and run tests against it (I'd prefer to avoid network usage to S3 every time tests run)

**Comment by cgwalters** [comment]
Ok, fixup ⬆️

**Comment by giuseppe** [comment]
@rh-atomic-bot r+ 3465a8b

**Pull request closed without merging**